### ISSUE TITLE
Ability to provide your own whitespace removal strategy

### ISSUE THREAD

**Issue by lalibi** [Taritsyn/WebMarkupMin#27]
Is it possible (through configuration or otherwise) to instruct the minifier to keep the new-line characters? To be more specific, to replace any series of whitespase that contain at least one new-line character with a single new-line character.

For instance, I'd like this:

```html
<html>

  <head>
    <title>...</title>
  </head>

  <body>
     ...
  </body>

<html> 
```
to become:
```html
<html>
<head>
<title>...</title>
</head>
<body>
...
</body>
<html> 
```
and not: 
```html
<html><head><title>...</title></head><body> ... </body><html>
```
I think html is still readable this way (personally i prefer that), without major compromises concerning the final output size.

**Comment by Taritsyn** [comment]
Hello, Lalibi!

> Is it possible (through configuration or otherwise) to instruct the minifier to keep the new-line characters? To be more specific, to replace any series of whitespase that contain at least one new-line character with a single new-line character.

While this feature is not implemented, but in the future may appear (it will not be soon).

**Comment by Taritsyn** [comment]
Hello, Nicholaos!

This feature is implemented in [version 2.12.0](https://github.com/Taritsyn/WebMarkupMin/releases/tag/v2.12.0). Some of the functionality will be further developed (for example, integration with some CSS and JS minifiers).

**Comment by lalibi** [comment]
That's great news, thank you!
